### PR TITLE
fix: fixing partition_html to not include the script tag code

### PR DESCRIPTION
**Pull request by sparkbrains**
Added regex to eliminate script tags as well as strings in between.

### CODE REVIEW

File: unstructured/documents/xml.py [hunk 1]
```diff
@@ -50,6 +51,11 @@
         return super().pages
 
     def _read_xml(self, content):
+        """
+        The regex checks the script tags in content(string).
+        Remove text inside the script tag as well.
+        """
+        content = re.sub(r"<script[\s\S]*?(.*?)</script>", "", content, flags=re.IGNORECASE)
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply unstructured/documents/xml.py:58]: Have you looked at this regex warning from the automated scans?

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Please update the PR title so its formatted like `fix: <description-of-fix>`. Please also reference the issue in the PR description and add a testing section to the PR description with test instructions. Thanks for the contribution!

File: unstructured/documents/xml.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sparkbrains) [reply unstructured/documents/xml.py:58]: Yes I have seen the warning. But the regex does match those tags

**Comment by cragwolfe** [comment]
The CodeQL issue looks legit, see https://codeql.github.com/codeql-query-help/python/py-bad-tag-filter/#example , "Browsers will not only accept </script> as script end tags, but also tags such as </script foo="bar">" so we may as well make the regex handle that that and add a test case.

**Comment by MthwRobinson** [comment]
I'd also prefer just skipping script tags (and their child tags) when we iterate over the HTML tags versus substituting out the `<script>` content with the regex.

**Comment by sparkbrains** [comment]
okay I will make edits accordingly.

**Comment by sparkbrains** [comment]
I have fixed the CodeQL Issue. Can you check now?

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Instead of replacing the script tags with a regex, I'd prefer to skip the `script` tag as we iterate over the tags in the documents. You'd be able to accomplish that by skipping `script` tags and adding its descendants to `descendanttag_elems` here:

https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/documents/html.py#L101

**Comment by MthwRobinson** [comment]
Closing this PR. Per [this comment](https://github.com/Unstructured-IO/unstructured/issues/149#issuecomment-1432128327), the offending JS in the example document is coming in from a `td` and it appears we're already skipping `script` tags. Updated #149 accordingly

**Pull request closed without merging**